src/HW_models/NRF_EGU.h

```diff
@@ -8,12 +8,24 @@
 #define _NRF_HW_MODEL_EGU_H
 
 #include "nrfx.h"
+#include "hal/nrf_egu.h"
 
 #ifdef __cplusplus
 extern "C"{
 #endif
 
-extern NRF_EGU_Type NRF_EGU_regs[6];
+extern NRF_EGU_Type NRF_EGU_regs[EGU_COUNT];
+
+#define reg_to_id(p_reg)    ((size_t)p_reg - (size_t)(&NRF_EGU_regs[0])) / sizeof(NRF_EGU_Type)
+#define reg_to_irq(p_reg)   reg_to_id(p_reg) + SWI0_EGU0_IRQn
+#define task_to_id(task)    (task - NRF_EGU_TASK_TRIGGER0) / sizeof(NRF_EGU_regs[0].TASKS_TRIGGER[0])
```

Comment:
no users?

Reply:
Nice catch, it had to slip my attention during refactoring

Reply:
When I implemented support for PPI, I had to use REG_TO_ID and TASK_TO_ID, so I left it in public header.